### PR TITLE
Replaces extools with auxtools; atmos overhaul

### PR DESCRIPTION
**Pull request by Putnam3145**
## About The Pull Request

Extools has been replaced with auxtools, a Rust-based replacement.

LINDA has been mostly torn out and replaced with my own FEA follow-up, FDA, a highly parallel finite difference method atmospheric protocol which is far more consistent, fast and reasonable than the previous. It moves gas in a LINDA-like way, but never gets stuck, has heat properly moving around and has the same performance no matter how many pressure differences there are (yes, really). It doesn't break planetary atmos and is, in general, easier to understand.

## Why It's Good For The Game

Extools has been a bit of a disaster. It causes issues with planet atmos, it causes crashes when loading away missions, and is just... unsafe code. The move to Rust and designing of an atmos subsystem with it being written in a fast, safe, concurrent-able programming language to begin should bring a huge improvement.

## Changelog
:cl:
add: FDA, LINDA without the bookkeeping.
add: Putnamos, a simpler replacement for Monstermos. If I can get Monstermos to work, this will be Monstermos instead.
del: LINDA, the old active-turfs-based atmos subsystem.
del: Monstermos? I would rather not get rid of this, but I can't get it to work correctly.
server: Extools has been removed, and loading extools alongside auxtools will cause massive problems. If this is tested or merged, remove extools from the static files.
/:cl:

### CODE REVIEW

File: code/__DEFINES/subsystems.dm [hunk 1]
```diff
@@ -197,6 +198,19 @@
 
 #define RUNLEVELS_DEFAULT (RUNLEVEL_SETUP | RUNLEVEL_GAME | RUNLEVEL_POSTGAME)
 
+// SSair run section
+#define SSAIR_PIPENETS 1
```
Comment (MarinaGryphon) [inline code/__DEFINES/subsystems.dm:204]: Lines 245-254 need to be removed.

**Review by MarinaGryphon — CHANGES_REQUESTED** [review]
code\__HELPERS\_extools_api.dm is not included.

File: code/controllers/subsystem/time_track.dm [hunk 2]
```diff
@@ -16,33 +17,81 @@ SUBSYSTEM_DEF(time_track)
...
+			SSair.cost_hotspots,
+			SSair.cost_superconductivity,
+			SSair.cost_pipenets,
+			SSair.cost_rebuilds,
+			SSair.get_active_turfs(), //does not return a list, which is what we want
+			SSair.get_amt_excited_groups(),
+			length(SSair.hotspots),
+			length(SSair.networks),
+			length(SSair.high_pressure_delta),
+			length(SSair.active_super_conductivity)
```
Comment (MarinaGryphon) [inline code/controllers/subsystem/time_track.dm]:
```suggestion
```

File: code/controllers/subsystem/time_track.dm [hunk 3]
```diff
@@ -16,33 +17,81 @@ SUBSYSTEM_DEF(time_track)
...
+			length(SStimer.timer_id_dict),
+			SSair.cost_turfs,
+			SSair.cost_groups,
+			SSair.cost_highpressure,
+			SSair.cost_hotspots,
+			SSair.cost_superconductivity,
+			SSair.cost_pipenets,
+			SSair.cost_rebuilds,
+			SSair.get_active_turfs(), //does not return a list, which is what we want
+			SSair.get_amt_excited_groups(),
```
Comment (MarinaGryphon) [inline code/controllers/subsystem/time_track.dm]:
```suggestion
```

File: code/controllers/subsystem/time_track.dm [hunk 4]
```diff
@@ -16,33 +17,81 @@ SUBSYSTEM_DEF(time_track)
...
+			MAPTICK_LAST_INTERNAL_TICK_USAGE,
+			length(SStimer.timer_id_dict),
+			SSair.cost_turfs,
+			SSair.cost_groups,
+			SSair.cost_highpressure,
+			SSair.cost_hotspots,
+			SSair.cost_superconductivity,
+			SSair.cost_pipenets,
+			SSair.cost_rebuilds,
+			SSair.get_active_turfs(), //does not return a list, which is what we want
```
Comment (MarinaGryphon) [inline code/controllers/subsystem/time_track.dm]:
```suggestion
```
These removed entries may need to be replaced with some of the new procs/vars, like `turf_process_time()`, `heat_process_time()`, `cost_post_process`, etc., depending on how logs are formatted.

**Review by MarinaGryphon — CHANGES_REQUESTED** [review]
~~I found out a way to leave suggestions on unchanged lines too.~~

File: code/game/world.dm [hunk 5]
```diff
@@ -7,46 +7,25 @@ GLOBAL_VAR_INIT(tgs_initialized, FALSE)
...
+		enable_debugging()
+	AUXTOOLS_CHECK(AUXMOS)
 #ifdef REFERENCE_TRACKING
 	enable_reference_tracking()
 #endif
-
 	world.Profile(PROFILE_START)
+	AUXTOOLS_CHECK(AUXMAPTICK)
+	initialize_maptick()
+	log_world("World loaded at [TIME_STAMP("hh:mm:ss", FALSE)]!")
```
Comment (MarinaGryphon) [inline code/game/world.dm:23]: This is a duplicate of line 32.

File: code/__DEFINES/is_helpers.dm [hunk 6]
```diff
@@ -240,3 +240,5 @@ GLOBAL_LIST_INIT(glass_sheet_types, typecacheof(list(
 #define isshuttleturf(T) (length(T.baseturfs) && (/turf/baseturf_skipover/shuttle in T.baseturfs))
 
 #define isProbablyWallMounted(O) (O.pixel_x > 20 || O.pixel_x < -20 || O.pixel_y > 20 || O.pixel_y < -20)
+
+#define isfinite(n) (isnum(n) && n == n)
```
Comment (MarinaGryphon) [inline code/__DEFINES/is_helpers.dm:258]: This doesn't actually work.
```suggestion
#define isfinite(n) (x != 1.#IND && abs(x) < 1.#INF)
```
Handles NaN, positive infinity, and negative infinity.

**Review by MarinaGryphon — CHANGES_REQUESTED** [review]
I made a mistake with the `isfinite` logic when testing it. It straight up doesn't work as-is.

**Comment by DeltaFire15** [comment]
Merge conflicting.

**Comment by Putnam3145** [comment]
(this is gonna fail until the CI is updated to 514)

**Comment by DeltaFire15** [comment]
Merge conflicting once again.

**Comment by DeltaFire15** [comment]
Marking this PR as stale due to its merge conflicting state combined with having gone a long timeframe without commits.

**Comment by MitztheKat** [comment]
If you added this to default clones of citadel... you are the sole cause of my aneurysm today. 
I spent six hours trying to run citadel without knowing this exists

**Comment by MarinaGryphon** [comment]
> If you added this to default clones of citadel... you are the sole cause of my aneurysm today.
> I spent six hours trying to run citadel without knowing this exists

You're misunderstanding. Citadel has extools-based atmos, currently, which only works below a certain 513 BYOND version. This PR *fixes* that by replacing it with something that works on current versions and isn't a pain in the ass to update to newer ones.

**Comment by silicons** [comment]
i love you so much putnam

**Pull request merged**